src/profile/components/multiItemEditor/MultiItemEditor.tsx

```diff
@@ -0,0 +1,173 @@
+        listAriaLabel: t('profileInformation.ariaListTitlePhones'),
+        listNumberTitle: t('profileInformation.phone'),
+        addNew: t('profileForm.addAnotherPhone'),
+        noContent: t('profileInformation.noPhones'),
+      };
+    }
+    return {
+      modalTitle: t('confirmationModal.removeAddress'),
+      title: t('profileInformation.addresses'),
+      listAriaLabel: t('profileInformation.ariaListTitleAddresses'),
+      listNumberTitle: t('profileInformation.address'),
+      addNew: t('profileForm.addAnotherAddress'),
+      noContent: t('profileInformation.noAddresses'),
+    };
+  })();
+
+  const executeActionAndNotifyUser: ActionListener = async (
+    action,
+    item,
+    newValue
+  ) => {
+    const [err] = await to(save(item, newValue as EditDataValue));
+    if (err) {
+      setErrorMessage(action);
+      return Promise.reject(err);
+    }
+    setSuccessMessage(action);
+    return Promise.resolve();
+  };
+
+  const onAction: ActionListener = async (action, item, newValue) => {
+    clearMessage();
+    if (action === 'cancel') {
+      if (isNewItem(item)) {
+        await remove(item);
+      } else {
+        reset(item);
+      }
+    } else if (action === 'save') {
+      return executeActionAndNotifyUser(action, item, newValue);
+    }
+    return Promise.resolve();
+  };
+
+  const NoItemsMessage = () => (
+    <React.Fragment>
+      {dataType === 'addresses' && (
+        <h3 className={commonFormStyles.sectionTitle}>
+          {texts.listNumberTitle}
+        </h3>
+      )}
+      <p>{texts.noContent}</p>
+    </React.Fragment>
+  );
+
+  return (
+    <ProfileSection>
+      <h3
+        className={classNames([
+          commonFormStyles.sectionTitle,
+          hasAddressList && commonFormStyles.visuallyHidden,
+        ])}
+      >
+        {texts.title}
+      </h3>
+
+      {!editDataList || (!editDataList.length && <NoItemsMessage />)}
+      <ul aria-label={texts.listAriaLabel} className={commonFormStyles.list}>
+        {editDataList.map((item, index) => (
+          <li
+            className={commonFormStyles.listItem}
+            aria-label={`${texts.listNumberTitle} ${index + 1}`}
+            key={item.id || 'new'}
+          >
+            <RowComponent
+              data={item}
+              onAction={onAction}
+              testId={`${dataType}-${index}`}
+              dataType={dataType}
+            />
+          </li>
+        ))}
+      </ul>
+      <EditingNotifications content={content} dataType={dataType} />
+      <Button
+        iconLeft={<IconPlusCircle />}
+        onClick={async () => {
+          clearMessage();
+          add();
+        }}
+        variant="secondary"
+        disabled={isAddButtonDisabled}
+        className={commonFormStyles.responsiveButton}
+        id={`${dataType}-add-button`}
+      >
+        {texts.addNew}
+      </Button>
+    </ProfileSection>
+  );
+}
+
+export default MultiItemEditor;
```

Comment:
Should we use the inline export instead of this?

Reply:
Following other exports in .tsx-files here

Reply:
Ok, sorry I thought that inline export was the convention.